doobie/src/main/scala/org/atnos/eff/addon/doobie/DoobieConnectionIOEffect.scala

```diff
@@ -58,6 +58,44 @@ trait DoobieConnectionIOInterpretation extends DoobieConnectionIOTypes {
       interceptErrors(before >> runEffect(connection) << after)(oops).addLast(send(always))
     }
   }
+
+  def runConnectionIO[R, U, F[_], E, A, B](e: Eff[R, A])(t: Transactor[F])(
+    implicit mc: Member.Aux[ConnectionIO, R, U],
+             mf: F /= U,
+             me: MonadError[F, E] ): Eff[U, A] = {
+
+    def getConnection: Eff[U, Connection] =
+      send[F, U, Connection](t.connect(t.kernel))
+
+    def interceptErrors[Y](effect: Eff[U, Y])(oops: F[Unit]): Eff[U, Y] =
+      interpret.interceptNat(effect)(new (F ~> F) {
+        def apply[X](f: F[X]): F[X] =
+          f.handleErrorWith((err: E) => oops *> me.raiseError[X](err))
+      })
+
+    interpret.translate(e)(new Translate[ConnectionIO, U] {
```

Comment:
Is there a way to factor out that code that's similar to the above method into one place?

Reply:
Definitely. Will update shortly.

Reply:
Was able to simplify this a lot and remove the duplicated code.